sig-ml-developer-experience/OWNERS

```diff
@@ -1,13 +1,20 @@
 approvers:
+- andrewballantyne
 - erwangranger
+- harshad16
 - jkoehler-redhat
 - kywalker-rh
```

Comment:
Best drop Erwan and Jay as well. I completely forgot about this file when I did my last pass.